src/sagemaker/multidatamodel.py

```diff
@@ -0,0 +1,211 @@
+        # Models trained with FrameworkEstimator set framework specific environment variables
+        # which need to be copied over
+        if self.model:
+            container_definition = self.model.prepare_container_def(instance_type, accelerator_type)
+            self.image = self.image or container_definition["Image"]
+            self.env.update(container_definition["Environment"])
+        return sagemaker.container_def(
+            self.image,
+            env=self.env,
+            model_data_url=self.model_data_prefix,
+            container_mode=self.container_mode,
+        )
+
+    def add_model(self, s3_url, model_data_path=None):
+        """Adds a model to the `MultiDataModel` by copying
+        the s3_url model artifact to the given S3 path relative to model_data_prefix
+
+        Args:
+        s3_url: S3 path of the trained model artifact
+        model_data_path: S3 path where the trained model artifact
+                should be uploaded relative to `self.model_data_prefix`.
+                (default: None). If None, then the entire s3_url path
+                (after the source bucketname) will be copied to
+                `model_data_prefix` location
+        """
+        # Validate s3_url
+        if not s3_url.startswith("s3://"):
+            raise ValueError(
+                'Expecting S3 model path beginning with "s3://". Received: "{}"'.format(s3_url)
+            )
+
+        source_bucket, source_model_data_path = self._parse_s3_uri(s3_url)
+        copy_source = {"Bucket": source_bucket, "Key": source_model_data_path}
+
+        if not model_data_path:
+            model_data_path = source_model_data_path
+
+        # Construct the destination path
+        dst_url = os.path.join(self.model_data_prefix, model_data_path)
+        destination_bucket, destination_model_data_path = self._parse_s3_uri(dst_url)
+
+        # Copy the model artifact
+        self.s3_client.copy(copy_source, destination_bucket, destination_model_data_path)
+
+    def list_models(self):
+        """Generates and returns relative paths to model archives stored at model_data_prefix
+        S3 location.
+
+        Yields: Relative paths to model archives stored at model_data_prefix.
+        """
+        bucket, url_prefix = self._parse_s3_uri(self.model_data_prefix)
+        paginator = self.s3_client.get_paginator("list_objects_v2")
+        for page in paginator.paginate(Bucket=bucket, Prefix=url_prefix):
+            try:
+                # Get the metadata about each object returned
+                s3_objects_metadata = page["Contents"]
+                for s3_object in s3_objects_metadata:
+                    # Return the model paths relative to the model_data_prefix
+                    # Ex: "a/b/c.tar.gz" -> "b/c.tar.gz" if url_prefix = "a/"
+                    yield s3_object["Key"].replace(url_prefix, "")
+            except KeyError:
+                return
+
+    def _parse_s3_uri(self, s3_url):
```

Comment:
This method can go into: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/s3.py with a different name

Reply:
Replaced this with `s3.parse_s3_url`